frontend/src/pages/projects/screens/detail/connections/ManageConnectionsModal.tsx

```diff
@@ -0,0 +1,176 @@
+        setNameDescData('description', previousEntries[type.metadata.name].nameDesc.description);
+        setConnectionValues(previousEntries[type.metadata.name].values);
+      } else {
+        // first time load, so add default values
+        const defaults: {
+          [key: string]: ConnectionTypeValueType;
+        } = {};
+        for (const field of type?.data?.fields ?? []) {
+          if (isConnectionTypeDataField(field) && field.properties.defaultValue) {
+            defaults[field.envVar] = field.properties.defaultValue;
+          }
+        }
+        setConnectionValues(defaults);
+      }
+
+      setSelectedConnectionType(type);
+    },
+    [selectedConnectionType, previousEntries, nameDescData, connectionValues, setNameDescData],
+  );
+
+  const isValid = React.useMemo(
+    () =>
+      !!selectedConnectionType &&
+      !!nameDescData.name &&
+      !selectedConnectionType.data?.fields?.find(
+        (field) =>
+          isConnectionTypeDataField(field) &&
+          field.required &&
+          !connectionValues[field.envVar] &&
+          field.type !== ConnectionTypeFieldType.Boolean,
+      ),
+    [selectedConnectionType, nameDescData, connectionValues],
+  );
+
+  return (
+    <Modal
+      title="Add Connection"
+      isOpen
+      onClose={() => {
+        onClose();
+      }}
+      variant="medium"
+      footer={
+        <DashboardModalFooter
+          submitLabel="Create"
+          onCancel={onClose}
+          onSubmit={() => {
+            const connectionValuesAsStrings = Object.fromEntries(
+              Object.entries(connectionValues).map(([key, value]) => [key, String(value)]),
+            );
+            const update: Connection = {
+              apiVersion: 'v1',
+              kind: 'Secret',
+              metadata: {
+                name: nameDescData.k8sName.value || translateDisplayNameForK8s(nameDescData.name),
+                namespace: project?.metadata.name ?? '',
+                labels: {
+                  'opendatahub.io/dashboard': 'true',
+                  'opendatahub.io/managed': 'true',
+                },
+                annotations: {
+                  'opendatahub.io/connection-type': selectedConnectionType?.metadata.name ?? '',
+                  'openshift.io/display-name': nameDescData.name,
+                  'openshift.io/description': nameDescData.description,
+                },
+              },
+              stringData: connectionValuesAsStrings,
+            };
+
+            setIsSaving(true);
+            setError(undefined);
+
+            onSubmit(update)
+              .then(() => {
+                onClose(true);
+              })
+              .catch((e) => {
+                setError(e);
+                setIsSaving(false);
+              });
+          }}
+          error={error}
+          isSubmitDisabled={!isValid}
+          isSubmitLoading={isSaving}
+          alertTitle=""
+        />
+      }
+    >
+      <ConnectionTypePreview
```

Comment:
This shouldn't be named `ConnectionTypePreview` anymore since it's being reused here and the preview. 
Maybe `ConnectionTypeForm`